### PR TITLE
refractor(df/locale):products module

### PR DESCRIPTION
**Pull request by abhishekkumar08**
## Description
Update dateformats and language with centralized data in products
module

## Related issues and discussion
#1213 

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

**Comment by BLasan** [comment]
@abhishekkumar08 Shall we rename the commit message as **refactor: update dateformats and language with centralized data in products module** . No need of two lines for a commit message in here

**Comment by abhishekkumar08** [comment]
@BLasan okay would do that. Thanks!

File: src/app/products/charges/create-charge/create-charge.component.ts [hunk 1]
```diff
@@ -43,12 +44,14 @@ export class CreateChargeComponent implements OnInit {
...
    * @param {Router} router Router for navigation.
    * @param {DatePipe} datePipe Date Pipe to format date.
+   * @param {SettingsService} settingsService Settings Service
    */
   constructor(private formBuilder: FormBuilder,
-              private productsService: ProductsService,
-              private route: ActivatedRoute,
-              private router: Router,
-              private datePipe: DatePipe) {
+    private productsService: ProductsService,
```
Comment (BLasan) [inline src/app/products/charges/create-charge/create-charge.component.ts:50]: Make vertical alignments of variables consistent to line 49 variable.

File: src/app/products/charges/edit-charge/edit-charge.component.ts [hunk 2]
```diff
@@ -43,18 +45,20 @@ export class EditChargeComponent implements OnInit {
...
+  /**
+   * Retrieves the charge data from `resolve`.
+   * @param {ProductsService} productsService Products Service.
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {Router} router Router for navigation.
+   * @param {SettingsService} settingsService Settings Service
+   */
   constructor(
     private productsService: ProductsService,
```
Comment (BLasan) [inline src/app/products/charges/edit-charge/edit-charge.component.ts:57]: Make proper vertical alignment as in other components

File: src/app/products/fixed-deposit-products/create-fixed-deposit-product/create-fixed-deposit-product.component.ts [hunk 3]
```diff
@@ -29,9 +30,14 @@ export class CreateFixedDepositProductComponent implements OnInit {
...
   accountingRuleData = ['None', 'Cash'];
 
+  /**
+   * @param {SettingsService} settingsService Settings Service.
+   */
+
   constructor(private route: ActivatedRoute,
-              private productsService: ProductsService,
-              private router: Router) {
+    private productsService: ProductsService,
```
Comment (BLasan) [inline src/app/products/fixed-deposit-products/create-fixed-deposit-product/create-fixed-deposit-product.component.ts:38]: vertical alignment of variables

File: src/app/products/fixed-deposit-products/edit-fixed-deposit-product/edit-fixed-deposit-product.component.ts [hunk 4]
```diff
@@ -29,9 +31,14 @@ export class EditFixedDepositProductComponent implements OnInit {
   fixedDepositProductsTemplate: any;
   accountingRuleData = ['None', 'Cash'];
 
+  /**
```
Comment (BLasan) [inline src/app/products/fixed-deposit-products/edit-fixed-deposit-product/edit-fixed-deposit-product.component.ts:34]: Add params for other variables declared within constructor

File: src/app/products/floating-rates/edit-floating-rate/edit-floating-rate.component.ts [hunk 5]
```diff
@@ -55,13 +56,15 @@ export class EditFloatingRateComponent implements OnInit {
...
    * @param {MatDialog} dialog Dialog reference.
+   * @param {SettingsService} settingsService Settings Service.
    */
   constructor(private router: Router,
-              private formBuilder: FormBuilder,
-              private productsService: ProductsService,
-              private route: ActivatedRoute,
-              private datePipe: DatePipe,
-              private dialog: MatDialog) {
+    private formBuilder: FormBuilder,
```
Comment (BLasan) [inline src/app/products/floating-rates/edit-floating-rate/edit-floating-rate.component.ts:62]: Consider vertical alignment as mentioned above. This should be applicable for all the components you have modified.

File: src/app/products/products.service.ts [hunk 6]
```diff
@@ -105,10 +111,10 @@ export class ProductsService {
   }
 
   getDividendData(shareProductId: any, dividendId: any): Observable<any> {
-    const httpParams = new HttpParams().set('dateFormat', 'dd MMMM yyyy')
-                                        .set('limit', '10')
-                                        .set('locale', 'en')
-                                        .set('offset', '0');
+    const httpParams = new HttpParams().set('dateFormat', this.settingsService.dateFormat)
```
Comment (BLasan) [inline src/app/products/products.service.ts:114]: Consider the alignments

**Review by BLasan — CHANGES_REQUESTED** [review]
@abhishekkumar08 Please make the requested changes

**Comment by abhishekkumar08** [comment]
@BLasan Thanks for the review would do that.

**Comment by abhishekkumar08** [comment]
@BLasan Please check if everything is in place. Should I squash the commits into one?

**Comment by BLasan** [comment]
> @BLasan Please check if everything is in place. Should I squash the commits into one or is it ok?

You need to squash the commits

**Comment by abhishekkumar08** [comment]
> 
> 
> > @BLasan Please check if everything is in place. Should I squash the commits into one or is it ok?
> 
> You need to squash the commits

Okay on that! Also everything's in place? Thanks for the review.

**Comment by BLasan** [comment]
> > > @BLasan Please check if everything is in place. Should I squash the commits into one or is it ok?
> > 
> > 
> > You need to squash the commits
> 
> Okay on that! Also everything's in place? Thanks for the review.

Will check :)

File: src/app/products/charges/edit-charge/edit-charge.component.ts [hunk 7]
```diff
@@ -43,18 +45,19 @@ export class EditChargeComponent implements OnInit {
...
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {Router} router Router for navigation.
+   * @param {SettingsService} settingsService Settings Service
+   */
+  constructor(private productsService: ProductsService,
+              private formBuilder: FormBuilder,
+              private route: ActivatedRoute,
+              private router: Router,
+              private settingsService: SettingsService
   ) {
```
Comment (BLasan) [inline src/app/products/charges/edit-charge/edit-charge.component.ts:61]: move these parentheses to line no 60

File: src/app/products/fixed-deposit-products/create-fixed-deposit-product/create-fixed-deposit-product.component.ts [hunk 8]
```diff
@@ -10,6 +10,7 @@ import { FixedDepositProductChargesStepComponent } from '../fixed-deposit-produc
 import { FixedDepositProductAccountingStepComponent } from '../fixed-deposit-product-stepper/fixed-deposit-product-accounting-step/fixed-deposit-product-accounting-step.component';
```
Comment (BLasan) [inline src/app/products/fixed-deposit-products/create-fixed-deposit-product/create-fixed-deposit-product.component.ts:10]: Shall we add headers(comments) for the imports?  You can get an idea by looking at edit-charge.component.ts file

**Comment by abhishekkumar08** [comment]
@BLasan okay, review all the code I will update the changes at last. @karantakalkar Please review this PR. Thanks.

File: src/app/products/share-products/share-product-stepper/share-product-market-price-step/share-product-market-price-step.component.ts [hunk 9]
```diff
@@ -23,9 +24,17 @@ export class ShareProductMarketPriceStepComponent implements OnInit {
 
   displayedColumns: string[] = ['fromDate', 'shareValue', 'actions'];
```
Comment (BLasan) [inline src/app/products/share-products/share-product-stepper/share-product-market-price-step/share-product-market-price-step.component.ts:25]: add the description for the variable

**Review by BLasan — CHANGES_REQUESTED** [review]
@abhishekkumar08 Nice work! Please make the required changes

**Comment by BLasan** [comment]
@abhishekkumar08 Please close the outdated requests

**Comment by abhishekkumar08** [comment]
> 
> 
> @abhishekkumar08 Please close the outdated requests

Didn't get that? @BLasan

**Comment by BLasan** [comment]
> > @abhishekkumar08 Please close the outdated reviews
> 
> Didn't get that? @BLasan

Outdated conversations regarding to the review.
Screenshot from 2021-01-17 23-35-12

**Comment by abhishekkumar08** [comment]
@karantakalkar Please review this PR. Thanks.

**Review by Jov03 — APPROVED** [review]

**Pull request merged**